File: app/src/main/java/moe/tlaster/openween/core/model/directmessage/DirectMessageModel.java
package moe.tlaster.openween.core.model.directmessage;

import com.google.gson.annotations.SerializedName;

import moe.tlaster.openween.core.model.user.UserModel;

/**
 * Created by Tlaster on 2016/9/2.
 */
public class DirectMessageModel {
    @SerializedName("id")
    private long mID;
    @SerializedName("idstr")
    private String mIDStr;
    @SerializedName("created_at")
    private String mCreatedAt;
    @SerializedName("text")
    private String mText;
    @SerializedName("sender_id")
    private long mSenderID;
    @SerializedName("recipient_id")
    private long mRecipientID;
    @SerializedName("recipient")
    private UserModel mRecipient;
    @SerializedName("sender")
    private UserModel mSender;
    @SerializedName("sender_screen_name")
    private String mSenderScreenName;
    @SerializedName("recipient_screen_name")
    private String mRecipientScreenName;
    @SerializedName("att_ids")
    private long[] AttIDs = { 0, 0 };

    public long getID() {
        return mID;
    }

    public void setID(long ID) {
        mID = ID;
    }

    public String getIDStr() {
        return mIDStr;
    }

    public void setIDStr(String IDStr) {
        mIDStr = IDStr;
    }

    public String getCreatedAt() {
        return mCreatedAt;
    }

    public void setCreatedAt(String createdAt) {
        mCreatedAt = createdAt;
    }

    public String getText() {
        return mText;
    }

    public void setText(String text) {
        mText = text;
    }

    public long getSenderID() {
        return mSenderID;
    }

    public void setSenderID(long senderID) {
        mSenderID = senderID;
    }

    public long getRecipientID() {
        return mRecipientID;
    }

    public void setRecipientID(long recipientID) {
        mRecipientID = recipientID;
    }

    public UserModel getRecipient() {
        return mRecipient;
    }

    public void setRecipient(UserModel recipient) {
        mRecipient = recipient;
    }

    public UserModel getSender() {
        return mSender;
    }

    public void setSender(UserModel sender) {
        mSender = sender;
    }

    public String getSenderScreenName() {
        return mSenderScreenName;
    }

    public void setSenderScreenName(String senderScreenName) {
        mSenderScreenName = senderScreenName;
    }

    public String getRecipientScreenName() {
        return mRecipientScreenName;
    }

    public void setRecipientScreenName(String recipientScreenName) {
        mRecipientScreenName = recipientScreenName;
    }

    public long[] getAttIDs() {
        return AttIDs;
    }

    public void setAttIDs(long[] attIDs) {
        AttIDs = attIDs;
    }
}

File: app/src/main/java/moe/tlaster/openween/core/model/directmessage/DirectMessageModel.kt
package moe.tlaster.openween.core.model.directmessage



import com.fasterxml.jackson.annotation.JsonProperty
import moe.tlaster.openween.core.model.user.UserModel

/**
 * Created by Tlaster on 2016/9/2.
 */
class DirectMessageModel {
    @field:JsonProperty("id")
    var id: Long = 0
    @field:JsonProperty("idstr")
    var idStr: String? = null
    @field:JsonProperty("created_at")
    var createdAt: String? = null
    @field:JsonProperty("text")
    var text: String? = null
    @field:JsonProperty("sender_id")
    var senderID: Long = 0
    @field:JsonProperty("recipient_id")
    var recipientID: Long = 0
    @field:JsonProperty("recipient")
    var recipient: UserModel? = null
    @field:JsonProperty("sender")
    var sender: UserModel? = null
    @field:JsonProperty("sender_screen_name")
    var senderScreenName: String? = null
    @field:JsonProperty("recipient_screen_name")
    var recipientScreenName: String? = null
    @field:JsonProperty("att_ids")
    var attIDs = longArrayOf(0, 0)
}
